accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,165 @@
+
+Source generator conflict resolution should run before compile, but after all NuGet package assets have been evaluated.  Specific sequencing, public/private target naming, extensibility will not be specified here but should follow best practice as determined by the SDK team. 
+
+### Source generator versioning
+
+Source generators should be strongly named and update assembly version with every public stable release (GA and servicing).  This ensures that the publicly shipped source generators can be loaded side-by-side in the same process and follows our versioning guidelines for components which ship and must run on .NETFramework.  Source generators need not update assembly version during pre-release but should update file version to facilitate well behaved distribution by installers.  In general source generators need to follow all the same guidelines as other code that ships out of dotnet/runtime and runs outside the shared framework.
+
+### Source generator compatibility concerns
+
+Source generators should only generate code which uses public API in framework components they extend.  This public API already needs to meet the strict binary compatibility guarantees as framework API.  As such libraries containing source-generated code will be binary-compatible with future releases, just as if the user wrote this code directly.
+
+Source generators should not generate code which uses public API with "reserved" or "undocumented" functionality in attempts to avoid public API guarantees.  Libraries paired with source generators need to provide the same guarantees as libraries alone do today.
+
+Source generators should avoid emitting public API in a user's projects.  Such API can create a compatibility contract in that user's library which needs to be maintained.  If a source generator produces public API in a user's project by design, that public API must be strictly maintained following the same rules as framework components across versions of the source generator.  Care should be taken to the public API which is generated and should be reviewed by the Framework Design Council.  A similar constraint applies to internal API: users will have source that expects to interact with the generated internal API and the user will expect that code to continue to compile between versions.
+
+Generated code should not be generated in System namespaces, it should prefer the user's namespace.  Generated code should be triple-slashed commented and be free from compiler warnings.  Generated code should avoid name collisions with user code and provide affordances for resolving name conflicts.  As much as possible generated code should meet the quality statndards of .NET library code, using nullable annotations meeting framework design and naming guidelines, etc.  It should feel like a natural extension of the library.
```

Comment:
"using nullable annotations meeting framework design and naming guidelines, etc"

I feel like you meant that it should use nullable annotations and also follow design guidelines.  A comma would help, or flipping the clauses

"As much as possible, the generated code should meet the quality standards of .NET library code, adhere to framework design and naming guidelines, utilize nullable annotations, etc."

In my sentence I also fixed `statndards` to `standards` :smile: